src/cli.rs

```diff
@@ -1456,6 +1465,178 @@ mod tests {
         assert!(display.contains("Will flag with aoflagger"));
         assert!(display.contains("Will not correct geometry"));
     }
+
+    /// Middle channel rounded-down is DC flagged when `flag_dc` is set.
+    #[test]
+    fn test_dc_flagging_odd_channel_count() {
+        let (metafits_path, gpufits_paths) = get_1254670392_avg_paths();
+
+        #[rustfmt::skip]
+        let args = vec![
+            "birli",
+            "-m", metafits_path,
+            "--no-draw-progress",
+            "--emulate-cotter",
+            gpufits_paths[0],
+            gpufits_paths[1],
+        ];
+
+        let BirliContext {
+            flag_ctx,
+            vis_sel,
+            num_timesteps_per_chunk,
+            corr_ctx,
+            ..
+        } = BirliContext::from_args(&args).unwrap();
+
+        let chunk_size = if let Some(steps) = num_timesteps_per_chunk {
+            steps
+        } else {
+            vis_sel.timestep_range.len()
+        };
```

Comment:
As a Haskell person, you'll like this: this could probably be simplified to `num_timesteps_per_chunk.unwrap_or(vis_sel.timestep_range.len());`